lib/lib-openapi/src/test/java/org/sagebionetworks/translator/ControllerToControllerModelTranslatorTest.java

```diff
@@ -0,0 +1,682 @@
+
+		JsonSchema expectedSchema = new JsonSchema();
+		expectedSchema.setType(Type.integer);
+		expectedSchema.setFormat("int32");
+		ParameterModel expectedParameter = new ParameterModel().withDescription(PARAM_1_DESCRIPTION)
+				.withIn(ParameterLocation.path).withName(PARAM_1_NAME).withRequired(true).withSchema(expectedSchema);
+		expectedParameters.add(expectedParameter);
+
+		return expectedParameters;
+	}
+
+	public RequestBodyModel getExpectedRequestBody() {
+		JsonSchema expectedSchema = new JsonSchema();
+		expectedSchema.setType(Type.integer);
+		expectedSchema.setFormat("int32");
+		return new RequestBodyModel().withDescription(PARAM_1_DESCRIPTION).withRequired(true)
+				.withSchema(expectedSchema);
+	}
+
+	private ResponseModel getExpectedResponseModel() {
+		JsonSchema schema = new JsonSchema();
+		schema.setType(Type.integer);
+		schema.setFormat("int32");
+		return new ResponseModel().withDescription(METHOD_RETURN_COMMENT).withContentType("application/json")
+				.withStatusCode(200).withSchema(schema);
+	}
+
+	private List<MethodModel> getExpectedMethods() {
+		List<MethodModel> expectedMethods = new ArrayList<>();
+		MethodModel expectedMethod = new MethodModel().withPath(METHOD_PATH).withDescription(METHOD_BEHAVIOR_COMMENT)
+				.withName(METHOD_NAME).withOperation(Operation.get).withParameters(getExpectedParameters())
+				.withRequestBody(getExpectedRequestBody()).withResponse(getExpectedResponseModel());
+		expectedMethods.add(expectedMethod);
+		return expectedMethods;
+	}
+
+	private void addAnnotationElementValues(Map<ExecutableElement, AnnotationValue> elementValues, String keyName,
+			Object value) {
+		ExecutableElement key = Mockito.mock(ExecutableElement.class);
+		Name simpleName = Mockito.mock(Name.class);
+		when(key.getSimpleName()).thenReturn(simpleName);
+		when(simpleName.toString()).thenReturn(keyName);
+
+		AnnotationValue annotationValue = Mockito.mock(AnnotationValue.class);
+		when(annotationValue.getValue()).thenReturn(value);
+
+		elementValues.put(key, annotationValue);
+	}
+
+	@Test
+	public void testTranslate() {
+		ControllerModel expectedControllerModel = new ControllerModel().withDisplayName(CONTROLLER_NAME)
+				.withMethods(getExpectedMethods()).withPath("/");
+
+		TypeElement controller = Mockito.mock(TypeElement.class);
+		when(controller.toString()).thenReturn(CONTROLLER_NAME);
+
+		Mockito.doReturn(getExpectedMethods()).when(translator).getMethods(any(List.class), any(DocTrees.class));
+		// call under test
+		assertEquals(expectedControllerModel, translator.translate(controller, Mockito.mock(DocTrees.class)));
+	}
+
+	@Test
+	public void testGetMethods() {
+		DocTrees docTrees = Mockito.mock(DocTrees.class);
+		DocCommentTree mockDocCommentTree = Mockito.mock(DocCommentTree.class);
+		when(docTrees.getDocCommentTree(any(Element.class))).thenReturn(mockDocCommentTree);
+		when(mockDocCommentTree.getBlockTags()).thenReturn(new ArrayList<>());
```

Comment:
create a local tags list that is returned with this when.

then in the verify you can check that that was passed to the spy.